bundles/org.openhab.binding.icloud/src/main/java/org/openhab/binding/icloud/internal/handler/ICloudAccountBridgeHandler.java

```diff
@@ -86,21 +99,152 @@ public void handleCommand(ChannelUID channelUID, Command command) {
         }
     }
 
+    @SuppressWarnings("null")
     @Override
     public void initialize() {
         logger.debug("iCloud bridge handler initializing ...");
-        iCloudDeviceInformationCache = new ExpiringCache<>(CACHE_EXPIRY, () -> {
-            try {
-                return connection.requestDeviceStatusJSON();
-            } catch (IOException e) {
-                logger.warn("Unable to refresh device data", e);
-                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+
+        if (authState != AuthState.WAIT_FOR_CODE) {
+            authState = AuthState.INITIAL;
+        }
+
+        callApiWithRetryAndExceptionHandling(() -> {
+            logger.debug("Dummy call for initial authentication.");
+            return null;
+        });
+
+        this.iCloudDeviceInformationCache = new ExpiringCache<>(CACHE_EXPIRY, () -> {
+            return callApiWithRetryAndExceptionHandling(() -> {
+                // callApiWithRetryAndExceptionHanlding ensures that iCloudService is not null when the following is
+                // called. Cannot use method local iCloudService instance here, because instance may be replaced with a
+                // new
+                // one during retry.
+                return iCloudService.getDevices().refreshClient();
+            });
+
+        });
+
+        if (authState == AuthState.AUTHENTICATED) {
+            ICloudAccountThingConfiguration config = getConfigAs(ICloudAccountThingConfiguration.class);
+            this.refreshJob = this.scheduler.scheduleWithFixedDelay(this::refreshData, 0, config.refreshTimeInMinutes,
+                    MINUTES);
+        } else {
+            cancelRefresh();
+        }
+        logger.debug("iCloud bridge handler initialized.");
+    }
```

Comment:
Before leaving this method, bridge status should be set. Are we sure it is?

Reply:
I'm allmost certain that it is. I tested again and followed all code paths and did not see any path not leading to a status update